S3/Exceptions.py

```diff
@@ -127,7 +127,7 @@ def parse_error_xml(tree):
         if not error_node.tag == "Error":
             error_node = tree.find(".//Error")
         if error_node is not None:
-            for child in error_node.getchildren():
+            for child in iter(error_node):
```

Comment:
in the "for" loop, you can use directly error_node as it is iterable.
No need for iter() in that case.
Same thing with the for in your 2 other changes.

Reply:
Fixed